src/execution.rs

```diff
@@ -319,6 +340,37 @@ impl Execution {
     pub fn in_far_call(&self) -> bool {
         self.running_contexts.len() > 1
     }
+
+    pub fn snapshot(&self) -> ExecutionSnapshot {
+        ExecutionSnapshot {
+            default_aa_code_hash: self.default_aa_code_hash,
+            evm_interpreter_code_hash: self.evm_interpreter_code_hash,
+            flag_eq: self.flag_eq,
+            flag_gt: self.flag_gt,
+            flag_lt_of: self.flag_lt_of,
+            heaps: self.heaps.clone(),
+            hook_address: self.hook_address,
+            register_context_u128: self.register_context_u128,
+            registers: self.registers,
+            running_contexts: self.running_contexts.clone(),
+            tx_number: self.tx_number,
+            use_hooks: self.use_hooks,
+        }
```

Comment:
I think this is the same as:
```suggestion
        self.clone()
```